src/mailbox-state.c

```diff
@@ -726,8 +732,13 @@ int mailbox_state_set_flags(struct mailbox_view *view,
 
 		if (*atom == '\\') {
 			/* system flag */
-			if (mail_flag_parse(atom + 1) == 0)
+			if ((flag = mail_flag_parse(atom + 1)) == 0)
+				return -1;
+			if ((flag & MAIL_RECENT) != 0 && imap4rev2_enabled) {
+				i_error("Unexpectedly trying to set a RECENT"
+					 "flag, while IMAP4REV2 is enabled");
```

Comment:
Should clarify a bit what it's complaining about. This is the FLAGS reply, right? So e.g. `Mailbox FLAGS response contains \Recent flag, while IMAP4REV2 is enabled`